### DISCUSSION TITLE
Is the value of the `log_cli` ini option introspectable from `conftest.py`?

### DISCUSSION THREAD

**Discussion by adigitoleo** [pytest-dev/pytest#12702] [Q&A] [answered]
If not, why is that the case? My use case is that I handle "live logging" using a custom logging plugin and want to override `log_cli` so that live logs are not duplicated.

In particular, I'm looking to get the value of that option with something like:

```python
@pytest.hookimpl(trylast=True)
def pytest_configure(config):
    if config.option.log_cli_option:
        # Suppress default live logs, turn on my custom variant
```

**Comment by wimglenn** [comment, marked as answer]
Yes, take a look at [`config.getini`](https://docs.pytest.org/en/latest/reference/reference.html#pytest.Config.getini).

if your `pyproject.toml` has this:

```toml
[tool.pytest.ini_options]
log_cli = true
```

Then your `conftest.py` can have this:

```
import pytest

@pytest.hookimpl(trylast=True)
def pytest_configure(config):
    if config.getini("log_cli"):
        print("do my stuff...")
```

You may also be interested in [`config.getoption`](https://docs.pytest.org/en/latest/reference/reference.html#pytest.Config.getoption), for retrieving command-line options (these have precedence over ini options).

**Reply by adigitoleo** [reply]
Thanks, that's fantastic. If the CLI options have precedence, I guess I should also check for `-o log_cli=true`? Or is that just a way to set ini options from the cli?

**Reply by wimglenn** [reply]
That's just another way to set ini options directly at the cmdline (`-o` is for config file _override_) and it would still be retrieved with `config.getini`.  If there was a cmdline option `--log-cli` (which there isn't), then that would be retrieved with `config.getoption`.

Does that make sense?

**Reply by adigitoleo** [reply]
Thanks, yes that was exactly the info I needed.
